dpnp/dpnp_algo/dpnp_algo_arraycreation.pyx

```diff
@@ -190,31 +190,65 @@ cpdef utils.dpnp_descriptor dpnp_identity(n, result_dtype):
     return result
 
 
-# TODO this function should work through dpnp_arange_c
-cpdef tuple dpnp_linspace(start, stop, num, endpoint, retstep, dtype, axis):
-    cdef shape_type_c obj_shape = utils._object_to_tuple(num)
-    cdef utils.dpnp_descriptor result = utils_py.create_output_descriptor_py(obj_shape, dtype, None)
+cpdef utils.dpnp_descriptor dpnp_linspace(start, stop, num, dtype=None, device=None, usm_type=None, sycl_queue=None, endpoint=True, retstep=False, axis=0):
+    usm_type_alloc, sycl_queue_alloc = utils_py.get_usm_allocations([start, stop])
 
-    if endpoint:
-        steps_count = num - 1
-    else:
-        steps_count = num
+    if sycl_queue is None and device is None:
+        sycl_queue = sycl_queue_alloc
+    sycl_queue_normalized = dpnp.get_normalized_queue_device(sycl_queue=sycl_queue, device=device)
```

Comment:
In below example there are both `a` and `b` arrays allocated on the same queue and device:
```
dev = "opencl:gpu"
a = dpnp.arange(2, sycl_queue=dpctl.SyclQueue(dev))
b = dpnp.arange(2, sycl_queue=a.sycl_queue)
c = dpnp.linspace(a, b, num=2, device=dev)

a.sycl_queue, a.device
Out: (<dpctl.SyclQueue at 0x7f1c3412aac0>, Device(opencl:gpu:0))

b.sycl_queue, b.device
Out: (<dpctl.SyclQueue at 0x7f1c3412aac0>, Device(opencl:gpu:0))

c.sycl_queue, c.device
(<dpctl.SyclQueue at 0x7f1c2dbdb8c0>, Device(opencl:gpu:0))

```
but the resulting array `c` is allocating on the same device but different queue.

Reply:
It is not a problem.